lsm-tree.cabal

```diff
@@ -52,45 +52,103 @@ description:
 
   == Performance #performance#
 
-  The worst-case time and space complexities are given in [big-O notation](http://en.wikipedia.org/wiki/Big_O_notation).
-  The time cost of operations on LSM-trees is generally dominated by the number of disk I\/O actions.
-  As such, the worst-case complexity of basic operations refer to the number of disk I\/O actions.
-
-  TODO: Describe the time complexity of the basic operations.
+  The worst-case behaviour of the library is described using [big-O notation](http://en.wikipedia.org/wiki/Big_O_notation).
+  The documentation provides two measures of complexity:
+
+  *   The time complexity of operations is described in terms of the number of disk I\/O operations and referred to as the disk I\/O complexity.
+      In practice, the time of the operations on LSM-trees is dominated by the number of disk I\/O actions.
+  *   The space complexity of tables is described in terms of the in-memory size of an LSM-tree table.
+      Both the in-memory and on-disk size of an LSM-tree table scales linearly.
+      However, the in-memory size of an LSM-tree table is smaller than its on-disk size by a significant constant.
+      This is discussed in detail below, under [In-memory size of tables](#performance_size).
+
+  The complexities are described in terms of the following variables and constants:
+
+  *   The variable \(n\) refers to the number of /physical/ table entries.
+      A /physical/ table entry is any key–operation pair, e.g., @Insert k v@ or @Delete k@, whereas a /logical/ table entry is determined by all physical entries with the same key.
+      If the variable \(n\) is used to describe the complexity of an operation that involves multiple tables, it refers to the sum of all table entries.
+  *   The constant \(T\) refers to the size ratio of the table, which is a configuration parameter.
+
+  === Disk I\/O cost of operations #performance_time#
```

Comment:
Small thing: we're summarising the complexity here but it is also documented in the `Simple` module, so it seems likely that it would get out of sync in the future if we're not careful

Reply:
True.